### PR TITLE
Native driver for Optelec ALVA 6 displays with support for raw braille input and the Optelec protocool converter

### LINKED ISSUE #6731: support for Optelec convertcard
Hi. I have an braille wouager and a Optelec convertecard wich makes it possible to use braille voyager for newer computers. I am not expert, but as i understand it, the card convert  braille voyager and even other older braildisplays from Tieman to a newer protocols like alva.
However this card is working on jaws, Windows eyes, and other screenreaders. However nvda does not work with this card. I believe it is more then me who have this issue, a new brailledisplay is not the cheepest eithem you can buy. so support for this would be an good idea. I am not a programer, but i'm not believe this is an bigger task to fix. That could be integrated in that allready alva drivers wich is allready included in NVDA.

### LINKED ISSUE #7733: Native driver for Optelec braille displays
Is it possible to start work on native driver for Optelec displays such as BC6 series?

Here are some benefits:
- many users use these displays (BC640/680, BC640 Comfort (USB and also emulate them with protocol converter attached to Alva Satellite or Tieman Voyager displays.
- It would be possible to use advanced braille features which are coming to NVDA.

Unfortunately I don't have enough skills to develop it but if I can help you on the other way, please tell me.

Thanks,

### PR DESCRIPTION
**Pull request by LeonarddeR**
First of all, many thanks to @dkager for his testing and fixes.

### Link to issue number:
Closes #7733.
Fixes #6731

### Summary of the issue:
The current DLL based ALVA driver has the following limitations:

* No support for raw keyboard input that allows utilizing NVDA's input gesture system and braille input framework
* No support for the Optelec Protocol Converter that allows using older displays with this driver

### Description of how this pull request fixes the issue:
This pull requests implements a new native driver for Optelec 6 series (BC640, BC680, 640 comfort and Protocol Converter). In comparison with the old driver, this changes/adds/fixes the following.

* This driver is completely native, and native is good :) since it is thread safe and allows braille display detection in the near future (#1271)
* When the number of cells reported by the display changes (e.g. due to setting a split point), NVDA's internal cell count is updated accordingly.
* When the time of a device is out of sync by more than 5 seconds, the time is synchronized automatically. This comes in quite handy, as ALVA displays always tend to lose their sense of time when the battery is empty.
* Support for raw keyboard input. When disabling HID keyboard input with the t1+spEnter gesture, this will allow a user to input braille using NVDA's own braille input system.

I've looked briefly at the BRLTTY code to support ALVA satelite displays and similar, and from the current code base, it shouldn't be that difficult to incorporate basic support for these displays. Furthermore, as we added support for the protocol converter, this driver is no longer only for the BC6 series. Therefore, we decided to rename the driver from alvaBc6 to Alva. Old BC6 driver users will be automatically moved to ALVA.

### Testing performed:
Tested two ALVA BC640 devices both over USB and Bluetooth.

### Known issues with pull request:
* The protocol converter hasn't yet been tested. CC @Bamze66, @gregjozk . Also, we haven't been able to test a BC680 or 640 comfort yet.
* Raw input is known to be non-functional on firmware 2.4.0 and older, however, NVDA will clearly state that changing HID input is not possible.

### Change log entry:
* New features
	+ Added support for the Optelec protocol converter, a device that allows one to use Braille Voyager and Satelite displays using the ALVA BC6 communication protocol. (#6731)
	+ It is now possible to use braille input with an ALVA 640 Comfort braille display. (#7733)
		- NVDA's braille input functionality can be used with these as well as other BC6 displays with firmware 3.0.0 and above.

### How to contribute
You can download the new driver [here](https://raw.githubusercontent.com/leonardder/nvda/bc6-native/source/brailleDisplayDrivers/alva.py). This link will always be up to date with the most recent source of the driver. Install by downloading the alva.py file and placing it in NVDA's brailleDisplayDrivers directory, found under %appdata\nvda or under userConfig in a portable copy.
Run NVDA and select this Braille display by opening the NVDA menu then selecting Preferences, Braille Settings, then choosing "Optelec ALVA 6 series/protocol converter". Note that the old driver is still available. If your device is not connected, make sure that you first select the "no braille" option and restart NVDA, in order to have the old driver not claim the display for itself.

### CODE REVIEW

**Review by michaelDCurran — APPROVED** [review]
Is it possible to force raw input by default for models that support it? Or if not, perhaps the userGuide entry could be re-worded to be more of a warning. Users would probably expect by default that NVDA's braille input features just work. To me the userGuide reads more as if the user simply does not have to worry as its just not relevent.

**Comment by leonardder** [comment]
> Is it possible to force raw input by default for models that support it? 

I'm very reluctant to do this, as it changes a setting inside the display without the end user knowing that. It's not just an internal toggle for the driver.

> Or if not, perhaps the userGuide entry could be re-worded to be more of a warning. Users would probably expect by default that NVDA's braille input features just work. To me the userGuide reads more as if the user simply does not have to worry as its just not relevent.

That is something I can do for sure.


@@Bamze66, @gregjozk, would you be able to test this driver with the protocol converter before it lands in next?

File: user_docs/en/userGuide.t2t [hunk 1]
```diff
@@ -1910,9 +1910,12 @@ Alternatively, you can connect an older Optelec display, such as a Braille Voyag
 You do not need any specific drivers to be installed to use these displays.
 Just plug in the display and configure NVDA to use it.
 
+Note: NVDA might be unable to use an ALVA BC6 display over Bluetooth when it is paired using the ALVA Bluetooth utility.
+When you have paired your device using this utility and NVDA is unable to detect your device, we recomment you to pair your ALVA display the regular way using the Windows Bluetooth settings.
```
Comment (michaelDCurran) [inline user_docs/en/userGuide.t2t]: typo: recomment -> recommend

**Review by michaelDCurran — CHANGES_REQUESTED** [review]

**Review by michaelDCurran — APPROVED** [review]

**Comment by michaelDCurran** [comment]
Can alvaw32.dll be removed from miscDeps now this is merged? I'm happy to do it, but just checking.

**Comment by dkager** [comment]
There is even a PR for that.

**Pull request merged**